src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -26,114 +26,123 @@
     @Autowired
     private ClinicService clinicService;
 
+
+    /**
+     * Method for parsing the csv and give a HTTP reply to the REST
+     *
+     * @param csv the csv string from the user
+     * @return ResponseEntity the HTTP response
+     */
     @PreAuthorize("hasRole(@roles.OWNER_ADMIN)")
     @RequestMapping(value = "importPets",
         method = RequestMethod.POST,
         consumes = "text/plain",
         produces = "application/json")
+
     public ResponseEntity<List<Pet>> importPets(@RequestBody String csv) {
 
-        int i = 0;
         List<Pet> pets = new LinkedList<Pet>();
-        Pet pet;
 
-        do {
-            pet = new Pet();
+        try{
+            pets = ParseFields(csv);
```

Comment:
as Ulrich said, this method ```ParseFields``` has more responsibilities/purposes than its name 
one function for parsing, one for error handling, one for adding/deleting